### PR TITLE
feat: Added support for Metal Interconnections

### PR DESCRIPTION
**Pull request by codinja1188**
## What this PR does / why we need it:
Added support for metal interconnections

## What issues has this PR fixed?
https://github.com/equinix/metal-cli/issues/307

### CODE REVIEW

File: internal/interconnections/interconnections.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	Servicer Servicer
+	Service  metal.InterconnectionsApiService
+	Out      outputs.Outputer
+}
+
+func (c *Client) NewCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     `interconnections`,
+		Aliases: []string{"conn"},
+		Short:   "interconnections operations: get.",
```
Comment (ctreatma) [inline internal/interconnections/interconnections.go]: Is `create` also supported?  I see a file for it

File: internal/interconnections/retrieve.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		Long:    "Retrieves a list of interconnections available to the default Organizations and by project ID",
+		Example: `  # Lists interconnections available to the default OrganizationID OR by project ID:	
+  metal interconnections get`,
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			var request metal.ApiOrganizationListInterconnectionsRequest
+
+			if projectID != "" {
+				request = c.Service.OrganizationListInterconnections(context.Background(), projectID)
```
Comment (ctreatma) [inline internal/interconnections/retrieve.go]: It looks like there are pagination attributes for this endpoint; we should add a pager here to make sure we load everything.

Comment (ctreatma) [reply internal/interconnections/retrieve.go]: Also it looks like this line is calling the wrong endpoint; should be calling the project interconnections endpoint.

File: internal/interconnections/retrieve.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+			} else {
+				request = c.Service.OrganizationListInterconnections(context.Background(), organizationID)
+			}
+
+			interConnectionsList, _, err := request.Execute()
+			if err != nil {
+				return fmt.Errorf("could not list metros: %w", err)
```
Comment (ctreatma) [inline internal/interconnections/retrieve.go]: error message references `metros`

File: docs/metal_interconnections_get.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+```
+metal interconnections get [flags]
+```
+
+### Examples
+
+```
+  # Lists interconnections available to the default OrganizationID OR by project ID:	
+  metal interconnections get
```
Comment (ctreatma) [inline docs/metal_interconnections_get.md]: This should document the flags for specifying an organization ID or a project ID

File: internal/interconnections/create.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+		Example: `  # Creates a new interconnection named "it-interconnection": 
+  metal interconnections create -n <name> [-m <metro>] [-r <redundancy>] [-t type ] [-p <project_id>] | [-O <organization_id>]`,
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			var request metal.ApiCreateOrganizationInterconnectionRequest
+			interconnectionCreateInput := metal.NewInterconnectionCreateInput(metro, name, redundancy, connType)
+
+			if projectID != "" {
+				request = c.Service.CreateOrganizationInterconnection(context.Background(), projectID).InterconnectionCreateInput(*interconnectionCreateInput)
```
Comment (ctreatma) [inline internal/interconnections/create.go]: This should be a CreateProjectInterconnection call

File: internal/interconnections/interconnections.go (hunk 1 above)
Comment (cprivitere) [reply internal/interconnections/interconnections.go]: Yeah looks like this is the text that gets used when you just do "metal help" and it's definitely missing the "create" command in that listing.

**Comment by cprivitere** [comment]
If I run this at all I get 404s:

```
> metal conn list
Error: Could not list Organization Interconnections: 404 Not Found
> metal conn create -n mynewInterconnection -m da -r primary -t shared
Error: could not create interconnections: 404 Not Found
```

File: internal/interconnections/create.go [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+			data[0] = []string{interconn.GetId(), interconn.GetName(), interconn.GetType(), interconn.GetCreatedAt().String()}
+			header := []string{"ID", "Name", "Type", "Created"}
+
+			return c.Out.Output(interconn, header, &data)
+		},
+	}
+
+	createInterconnectionsCmd.Flags().StringVarP(&name, "name", "n", "", "Name of the interconnection")
+	createInterconnectionsCmd.Flags().StringVarP(&metro, "metro", "m", "", "metro in the interconnection")
+	createInterconnectionsCmd.Flags().StringVarP(&redundancy, "redundancy", "r", "", "Website URL of the organization.")
```
Comment (cprivitere) [inline internal/interconnections/create.go:59]: The options for -r should be like single or redundant, not a website URL.

File: internal/interconnections/create.go [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+	createInterconnectionsCmd.Flags().StringVarP(&connType, "type", "t", "", "type of of interconnection.")
```
Comment (cprivitere) [inline internal/interconnections/create.go:60]: What are examples of the types?

**Comment by cprivitere** [comment]
I believe we're missing some stuff here, if you make a shared connection (VlanFabricVcCreateInput or VrfFabricVcCreateInput) you need to supply a "service_token_type" which I don't see referenced anywhere in this PR.

Here's the spec page I'm looking at:
https://deploy.equinix.com/developers/api/metal#tag/Interconnections/operation/createOrganizationInterconnection

**Comment by codinja1188** [comment]
> I believe we're missing some stuff here, if you make a shared connection (VlanFabricVcCreateInput or VrfFabricVcCreateInput) you need to supply a "service_token_type" which I don't see referenced anywhere in this PR.
> 
> Here's the spec page I'm looking at: https://deploy.equinix.com/developers/api/metal#tag/Interconnections/operation/createOrganizationInterconnection

we are supporting dedicated port input as of now

type: "dedicated"

File: internal/interconnections/retrieve.go [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+  metal interconnections get`,
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			var interConnections []metal.Interconnection
+			var proErr error
+			inc := []string{}
+			exc := []string{}
+			if projectID != "" {
+				interConnections, proErr = pager.GetAllProjectInterconnections(c.Service, projectID, inc, exc)
```
Comment (cprivitere) [inline internal/interconnections/retrieve.go]: I think we need a new flag in this file that lets people pass in another projectID if they don't want to only check their default project.

Comment (codinja1188) [reply internal/interconnections/retrieve.go]:
```
metal conn create -n "test-conn" -m da -r "primary"|"redundant" -t "dedicated"|"shared" -O <ORG_ID> -p <project_ID>
```

**Comment by cprivitere** [comment]
> ```
metal conn create -n "test-conn" -m da -r "primary"|"redundant" -t "dedicated"|"shared" -O <ORG_ID> -p <project_ID>
``` 

>  

That seems like it'd be right.

File: internal/interconnections/retrieve.go (hunk 8 above)
Comment (cprivitere) [reply internal/interconnections/retrieve.go]: That seems like it'd be right.

**Comment by cprivitere** [comment]
> > I believe we're missing some stuff here, if you make a shared connection (VlanFabricVcCreateInput or VrfFabricVcCreateInput) you need to supply a "service_token_type" which I don't see referenced anywhere in this PR.
> > Here's the spec page I'm looking at: https://deploy.equinix.com/developers/api/metal#tag/Interconnections/operation/createOrganizationInterconnection
> 
> we are supporting dedicated port input as of now
> 
> type: "dedicated"

Gotcha, so just -t "dedicated" for now.

File: internal/interconnections/create.go (hunk 7 above)
Comment (cprivitere) [reply internal/interconnections/create.go:60]: "shared" and "dedicated"

File: internal/interconnections/retrieve.go [hunk 9]
```diff
@@ -0,0 +1,91 @@
+// Copyright © 2023 Jasmin Gacic <jasmin@stackpointcloud.com>
```
Comment (ctreatma) [inline internal/interconnections/retrieve.go]: I don't think we need this license text at the top of every file

File: internal/interconnections/retrieve.go [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+			exc := []string{}
+			header := []string{"ID", "Name", "Code", "Type", "Created"}
+
+			if projectID != "" {
+				interConnections, proerr = pager.GetAllProjectInterconnections(c.Service, projectID, inc, exc)
+				if proerr != nil {
+					return fmt.Errorf("could not list Project Interconnections: %w", proerr)
+				}
+			} else if organizationID != "" {
+				interConnectionsList, _, err := c.Service.OrganizationListInterconnections(context.Background(), organizationID).Include(inc).Exclude(exc).Execute()
```
Comment (ctreatma) [inline internal/interconnections/retrieve.go:42]: Pagination is being handled for project interconnections now, but it looks like there is also pagination data available for OrganizationListInterconnections

**Comment by codinja1188** [comment]
# Delete

```
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn delete
Error: required flag(s) "id" not set
Usage:
  metal interconnections delete -i <connection_id>  [flags]

Examples:
  # Deletes the specified interconnection:
  metal interconnections delete -i 7ec86e23-8dcf-48ed-bd9b-c25c20958277

Flags:
  -h, --help        help for delete
  -i, --id string   The UUID of the interconnection.

Global Flags:
      --config string         Path to JSON or YAML configuration file
      --exclude strings       Comma separated Href references to collapse in results, may be dotted three levels deep
      --filter stringArray    Filter 'get' actions with name value pairs. Filter is not supported by all resources and is implemented as request query parameters.
      --http-header strings   Headers to add to requests (in format key=value)
      --include strings       Comma separated Href references to expand in results, may be dotted three levels deep
  -o, --output string         Output format (*table, json, yaml). env output formats are (*sh, terraform, capp).
      --search string         Search keyword for use in 'get' actions. Search is not supported by all resources.
      --sort-by string        Sort fields for use in 'get' actions. Sort is not supported by all resources.
      --sort-dir string       Sort field direction for use in 'get' actions. Sort is not supported by all resources.
      --token string          Metal API Token (METAL_AUTH_TOKEN)

Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn delete -i b090309f-0286-4586-84bd-3c349337a2c3
connection deletion initiated. Please check 'metal interconnections get -i b090309f-0286-4586-84bd-3c349337a2c3 ' for status
```

**Comment by codinja1188** [comment]
```
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn create -n "test-conn" -m da -r "primary" -t "dedicated" -p <Project_ID>
+--------------------------------------+-----------+-----------+-------------------------------+
|                  ID                  |   NAME    |   TYPE    |            CREATED            |
+--------------------------------------+-----------+-----------+-------------------------------+
| 3f508528-d8e3-429d-bbd8-b1c0ebb6a5f6 | test-conn | dedicated | 2023-10-12 14:00:33 +0000 UTC |
+--------------------------------------+-----------+-----------+-------------------------------+
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn create -n "test-conn" -m da -r "redundant" -t "dedicated" -p <Project_ID>
+--------------------------------------+-----------+-----------+-------------------------------+
|                  ID                  |   NAME    |   TYPE    |            CREATED            |
+--------------------------------------+-----------+-----------+-------------------------------+
| 7f75dc3c-d186-48bf-a983-32e01d74b607 | test-conn | dedicated | 2023-10-12 14:01:26 +0000 UTC |
+--------------------------------------+-----------+-----------+-------------------------------+
```

**Comment by codinja1188** [comment]
```
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn get -O 4d12f460-8c5e-43ea-986d-529d328815ee
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
|                  ID                  |                            NAME                            |   TYPE    |            CREATED            |
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
| 785fb586-9046-4d05-bee7-e8f752fa3033 | ansible-integration-test-connection-gnzlmmno-TestConn-vlan | shared    | 2023-10-05 22:31:38 +0000 UTC |
| f8c832de-ec40-4f74-b142-9a7c0e86caae | ansible-integration-test-connection-pzwqmfsd-TestConn-vlan | shared    | 2023-10-05 22:26:57 +0000 UTC |
| c84eaccc-b97f-4539-ac81-b58619ed3728 | ansible-integration-test-connection-gmzknjxd-TestConn-vlan | shared    | 2023-10-05 22:29:10 +0000 UTC |
| 3f508528-d8e3-429d-bbd8-b1c0ebb6a5f6 | test-conn                                                  | dedicated | 2023-10-12 14:00:33 +0000 UTC |
| 7f75dc3c-d186-48bf-a983-32e01d74b607 | test-conn                                                  | dedicated | 2023-10-12 14:01:26 +0000 UTC |
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn get -p 5211fb64-1e5b-402f-a904-f86427a5dc55
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
|                  ID                  |                            NAME                            |   TYPE    |            CREATED            |
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
| 785fb586-9046-4d05-bee7-e8f752fa3033 | ansible-integration-test-connection-gnzlmmno-TestConn-vlan | shared    | 2023-10-05 22:31:38 +0000 UTC |
| f8c832de-ec40-4f74-b142-9a7c0e86caae | ansible-integration-test-connection-pzwqmfsd-TestConn-vlan | shared    | 2023-10-05 22:26:57 +0000 UTC |
| c84eaccc-b97f-4539-ac81-b58619ed3728 | ansible-integration-test-connection-gmzknjxd-TestConn-vlan | shared    | 2023-10-05 22:29:10 +0000 UTC |
| 3f508528-d8e3-429d-bbd8-b1c0ebb6a5f6 | test-conn                                                  | dedicated | 2023-10-12 14:00:33 +0000 UTC |
| 7f75dc3c-d186-48bf-a983-32e01d74b607 | test-conn                                                  | dedicated | 2023-10-12 14:01:26 +0000 UTC |
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn get -i 785fb586-9046-4d05-bee7-e8f752fa3033
+--------------------------------------+------------------------------------------------------------+--------+-------------------------------+
|                  ID                  |                            NAME                            |  TYPE  |            CREATED            |
+--------------------------------------+------------------------------------------------------------+--------+-------------------------------+
| 785fb586-9046-4d05-bee7-e8f752fa3033 | ansible-integration-test-connection-gnzlmmno-TestConn-vlan | shared | 2023-10-05 22:31:38 +0000 UTC |
+--------------------------------------+------------------------------------------------------------+--------+-------------------------------+
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn create -n "test-conn" -m da -r "primary" -t "dedicated" -p 5211fb64-1e5b-402f-a904-f86427a5dc55
+--------------------------------------+-----------+-----------+-------------------------------+
|                  ID                  |   NAME    |   TYPE    |            CREATED            |
+--------------------------------------+-----------+-----------+-------------------------------+
| d17ce1ca-fedb-451f-af23-1e5611a3dbf3 | test-conn | dedicated | 2023-10-13 14:17:17 +0000 UTC |
+--------------------------------------+-----------+-----------+-------------------------------+
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn create -n "test-conn-1" -m da -r "primary" -t "shared" -p 5211fb64-1e5b-402f-a904-f86427a5dc55
Error: could not create interconnections: 422 Unprocessable Entity
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn create -n "test-conn-1" -m da -r "redundant" -t "dedicated" -p 5211fb64-1e5b-402f-a904-f86427a5dc55
+--------------------------------------+-------------+-----------+-------------------------------+
|                  ID                  |    NAME     |   TYPE    |            CREATED            |
+--------------------------------------+-------------+-----------+-------------------------------+
| 5423d90e-704d-45a4-a1dd-8c8c51ffd356 | test-conn-1 | dedicated | 2023-10-13 14:18:45 +0000 UTC |
+--------------------------------------+-------------+-----------+-------------------------------+
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn get -p 5211fb64-1e5b-402f-a904-f86427a5dc55
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
|                  ID                  |                            NAME                            |   TYPE    |            CREATED            |
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
| 785fb586-9046-4d05-bee7-e8f752fa3033 | ansible-integration-test-connection-gnzlmmno-TestConn-vlan | shared    | 2023-10-05 22:31:38 +0000 UTC |
| f8c832de-ec40-4f74-b142-9a7c0e86caae | ansible-integration-test-connection-pzwqmfsd-TestConn-vlan | shared    | 2023-10-05 22:26:57 +0000 UTC |
| c84eaccc-b97f-4539-ac81-b58619ed3728 | ansible-integration-test-connection-gmzknjxd-TestConn-vlan | shared    | 2023-10-05 22:29:10 +0000 UTC |
| 3f508528-d8e3-429d-bbd8-b1c0ebb6a5f6 | test-conn                                                  | dedicated | 2023-10-12 14:00:33 +0000 UTC |
| 7f75dc3c-d186-48bf-a983-32e01d74b607 | test-conn                                                  | dedicated | 2023-10-12 14:01:26 +0000 UTC |
| d17ce1ca-fedb-451f-af23-1e5611a3dbf3 | test-conn                                                  | dedicated | 2023-10-13 14:17:17 +0000 UTC |
| 5423d90e-704d-45a4-a1dd-8c8c51ffd356 | test-conn-1                                                | dedicated | 2023-10-13 14:18:45 +0000 UTC |
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn delete -i d17ce1ca-fedb-451f-af23-1e5611a3dbf3
connection deletion initiated. Please check 'metal interconnections get -i d17ce1ca-fedb-451f-af23-1e5611a3dbf3 ' for status
```

**Comment by codinja1188** [comment]
Test results:
```
Vasubabus-MacBook-Pro:metal-cli vasubabu$ make build
go build -ldflags "-X github.com/equinix/metal-cli/cmd.Version=c4dc1dc -X github.com/equinix/metal-cli/cmd.Build=c4dc1dc" -o bin/metal ./cmd/metal
Vasubabus-MacBook-Pro:metal-cli vasubabu$ cd bin/
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn get -p 5211fb64-1e5b-402f-a904-f86427a5dc55
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
|                  ID                  |                            NAME                            |   TYPE    |            CREATED            |
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
| 785fb586-9046-4d05-bee7-e8f752fa3033 | ansible-integration-test-connection-gnzlmmno-TestConn-vlan | shared    | 2023-10-05 22:31:38 +0000 UTC |
| f8c832de-ec40-4f74-b142-9a7c0e86caae | ansible-integration-test-connection-pzwqmfsd-TestConn-vlan | shared    | 2023-10-05 22:26:57 +0000 UTC |
| c84eaccc-b97f-4539-ac81-b58619ed3728 | ansible-integration-test-connection-gmzknjxd-TestConn-vlan | shared    | 2023-10-05 22:29:10 +0000 UTC |
| e9763728-d4d7-47de-b1ee-dc6504243327 | test-conn                                                  | dedicated | 2023-10-27 13:00:02 +0000 UTC |
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn update -i e9763728-d4d7-47de-b1ee-dc6504243327 -d "vasu testing interconnections"
+--------------------------------------+-----------+-----------+-------------------------------+
|                  ID                  |   NAME    |   TYPE    |            CREATED            |
+--------------------------------------+-----------+-----------+-------------------------------+
| e9763728-d4d7-47de-b1ee-dc6504243327 | test-conn | dedicated | 2023-10-27 13:00:02 +0000 UTC |
+--------------------------------------+-----------+-----------+-------------------------------+
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn get -p 5211fb64-1e5b-402f-a904-f86427a5dc55
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
|                  ID                  |                            NAME                            |   TYPE    |            CREATED            |
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
| 785fb586-9046-4d05-bee7-e8f752fa3033 | ansible-integration-test-connection-gnzlmmno-TestConn-vlan | shared    | 2023-10-05 22:31:38 +0000 UTC |
| f8c832de-ec40-4f74-b142-9a7c0e86caae | ansible-integration-test-connection-pzwqmfsd-TestConn-vlan | shared    | 2023-10-05 22:26:57 +0000 UTC |
| c84eaccc-b97f-4539-ac81-b58619ed3728 | ansible-integration-test-connection-gmzknjxd-TestConn-vlan | shared    | 2023-10-05 22:29:10 +0000 UTC |
| e9763728-d4d7-47de-b1ee-dc6504243327 | test-conn                                                  | dedicated | 2023-10-27 13:00:02 +0000 UTC |
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn create -n "vasu-interconn" -m da -r "primary" -t "dedicated" -p 5211fb64-1e5b-402f-a904-f86427a5dc55
+--------------------------------------+----------------+-----------+-------------------------------+
|                  ID                  |      NAME      |   TYPE    |            CREATED            |
+--------------------------------------+----------------+-----------+-------------------------------+
| 68282163-509b-4707-bb5d-4b0f5db5b594 | vasu-interconn | dedicated | 2023-10-27 13:32:48 +0000 UTC |
+--------------------------------------+----------------+-----------+-------------------------------+
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn get -p 5211fb64-1e5b-402f-a904-f86427a5dc55
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
|                  ID                  |                            NAME                            |   TYPE    |            CREATED            |
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
| 785fb586-9046-4d05-bee7-e8f752fa3033 | ansible-integration-test-connection-gnzlmmno-TestConn-vlan | shared    | 2023-10-05 22:31:38 +0000 UTC |
| f8c832de-ec40-4f74-b142-9a7c0e86caae | ansible-integration-test-connection-pzwqmfsd-TestConn-vlan | shared    | 2023-10-05 22:26:57 +0000 UTC |
| c84eaccc-b97f-4539-ac81-b58619ed3728 | ansible-integration-test-connection-gmzknjxd-TestConn-vlan | shared    | 2023-10-05 22:29:10 +0000 UTC |
| e9763728-d4d7-47de-b1ee-dc6504243327 | test-conn                                                  | dedicated | 2023-10-27 13:00:02 +0000 UTC |
| 68282163-509b-4707-bb5d-4b0f5db5b594 | vasu-interconn                                             | dedicated | 2023-10-27 13:32:48 +0000 UTC |
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn get -i 68282163-509b-4707-bb5d-4b0f5db5b594
+--------------------------------------+----------------+-----------+-------------------------------+
|                  ID                  |      NAME      |   TYPE    |            CREATED            |
+--------------------------------------+----------------+-----------+-------------------------------+
| 68282163-509b-4707-bb5d-4b0f5db5b594 | vasu-interconn | dedicated | 2023-10-27 13:32:48 +0000 UTC |
+--------------------------------------+----------------+-----------+-------------------------------+
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn get -o 4d12f460-8c5e-43ea-986d-529d328815ee
2023/10/27 19:03:24 unknown format: "4d12f460-8c5e-43ea-986d-529d328815ee". Using default.
+----+------+------+---------+
| ID | NAME | TYPE | CREATED |
+----+------+------+---------+
+----+------+------+---------+

Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn get -O 4d12f460-8c5e-43ea-986d-529d328815ee
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
|                  ID                  |                            NAME                            |   TYPE    |            CREATED            |
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
| 785fb586-9046-4d05-bee7-e8f752fa3033 | ansible-integration-test-connection-gnzlmmno-TestConn-vlan | shared    | 2023-10-05 22:31:38 +0000 UTC |
| f8c832de-ec40-4f74-b142-9a7c0e86caae | ansible-integration-test-connection-pzwqmfsd-TestConn-vlan | shared    | 2023-10-05 22:26:57 +0000 UTC |
| c84eaccc-b97f-4539-ac81-b58619ed3728 | ansible-integration-test-connection-gmzknjxd-TestConn-vlan | shared    | 2023-10-05 22:29:10 +0000 UTC |
| e9763728-d4d7-47de-b1ee-dc6504243327 | test-conn                                                  | dedicated | 2023-10-27 13:00:02 +0000 UTC |
| 68282163-509b-4707-bb5d-4b0f5db5b594 | vasu-interconn                                             | dedicated | 2023-10-27 13:32:48 +0000 UTC |
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+

Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn delete -i e9763728-d4d7-47de-b1ee-dc6504243327
connection deletion initiated. Please check 'metal interconnections get -i e9763728-d4d7-47de-b1ee-dc6504243327 ' for status
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal conn get -O 4d12f460-8c5e-43ea-986d-529d328815ee
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
|                  ID                  |                            NAME                            |   TYPE    |            CREATED            |
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------+
| 785fb586-9046-4d05-bee7-e8f752fa3033 | ansible-integration-test-connection-gnzlmmno-TestConn-vlan | shared    | 2023-10-05 22:31:38 +0000 UTC |
| f8c832de-ec40-4f74-b142-9a7c0e86caae | ansible-integration-test-connection-pzwqmfsd-TestConn-vlan | shared    | 2023-10-05 22:26:57 +0000 UTC |
| c84eaccc-b97f-4539-ac81-b58619ed3728 | ansible-integration-test-connection-gmzknjxd-TestConn-vlan | shared    | 2023-10-05 22:29:10 +0000 UTC |
| 68282163-509b-4707-bb5d-4b0f5db5b594 | vasu-interconn                                             | dedicated | 2023-10-27 13:32:48 +0000 UTC |
+--------------------------------------+------------------------------------------------------------+-----------+-------------------------------
```

**Review by displague — APPROVED** [review]

File: internal/interconnections/delete.go [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+		Use:   `delete -i <connection_id> `,
+		Short: "Deletes a interconnection.",
+		Long:  "Deletes the specified interconnection.",
+		Example: `  # Deletes the specified interconnection:
+  metal interconnections delete -i 7ec86e23-8dcf-48ed-bd9b-c25c20958277`,
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+
+			_, _, err := c.Service.DeleteInterconnection(context.Background(), connectionID).Execute()
```
Comment (displague) [inline internal/interconnections/delete.go:23]: Delete should require `--force` or an interactive prompt.

I'll open this as a new issue to keep this feature unblocked as-is.

**Pull request merged**